receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,311 @@
+		select {
+		case <-ctx.Done():
+			return
+		case <-s.stopperChan:
+			return
+		case <-time.After(collectionInterval):
+		}
+
+		for _, p := range s.cfg.Logs.Projects {
+			project, err := s.client.GetProject(ctx, p.Name)
+			if err != nil {
+				s.log.Error("Error retrieving project "+p.Name+":", zap.Error(err))
+				continue
+			}
+
+			org, err := s.client.GetOrganization(ctx, project.OrgID)
+			if err != nil {
+				s.log.Error("Error retrieving organization", zap.Error(err))
+				resource.Org = "unknown"
+			} else {
+				resource.Org = org.Name
+			}
+
+			// get clusters for each of the projects
+			resource.Project = *project
+			clusters, err := s.processClusters(ctx, *p, &resource)
+			if err != nil {
+				s.log.Error("Failure to process Clusters", zap.Error(err))
+			}
+
+			if !resource.end.IsZero() {
+				resource.start = resource.end
+			}
+			resource.end = time.Now()
+			s.collectClusterLogs(clusters, p, resource)
+		}
+	}
+}
+
+func (s *logsReceiver) processClusters(ctx context.Context, project Project, r *resourceInfo) ([]mongodbatlas.Cluster, error) {
+	include, exclude := project.IncludeClusters, project.ExcludeClusters
+	clusters, err := s.client.GetClusters(ctx, r.Project.ID)
+	if err != nil {
+		s.log.Error("Failure to collect clusters from project: %w", zap.Error(err))
+		return nil, err
+	}
+
+	// check to include or exclude clusters
+	switch {
+	// keep all clusters if include and exclude are not specified
+	case len(include) == 0 && len(exclude) == 0:
+		return clusters, nil
+	// include is initialized
+	case len(include) > 0 && len(exclude) == 0:
+		return filterClusters(clusters, createStringSet(include), true), nil
+	// exclude is initialized
+	case len(exclude) > 0 && len(include) == 0:
+		return filterClusters(clusters, createStringSet(exclude), false), nil
+	// both are initialized
+	default:
+		return nil, errors.New("both Include and Exclude clusters configured")
+	}
+}
+
+func (s *logsReceiver) collectClusterLogs(clusters []mongodbatlas.Cluster, project *Project, r resourceInfo) {
+	for _, cluster := range clusters {
+		hostnames := parseHostName(cluster.ConnectionStrings.Standard, s.log)
+		for _, hostname := range hostnames {
+			r.Cluster = cluster
+			r.Hostname = hostname
+			s.collectLogs(r, "mongodb.gz")
+			s.collectLogs(r, "mongos.gz")
+
+			if project.EnableAuditLogs {
+				s.collectAuditLogs(r, "mongodb-audit-log.gz")
+				s.collectAuditLogs(r, "mongos-audit-log.gz")
+			}
+		}
+	}
+}
+
+func filterClusters(clusters []mongodbatlas.Cluster, keys map[string]struct{}, include bool) []mongodbatlas.Cluster {
+	var filtered []mongodbatlas.Cluster
+	for _, cluster := range clusters {
+		if _, ok := keys[cluster.Name]; (!ok && !include) || (ok && include) {
+			filtered = append(filtered, cluster)
+		}
+	}
+	return filtered
+}
+
+func createStringSet(in []string) map[string]struct{} {
+	list := map[string]struct{}{}
+	for i := range in {
+		list[in[i]] = struct{}{}
+	}
+
+	return list
+}
```

Comment:
`createStringSet` is an efficiency optimization for `filterClusters`, which really means it only needs to be a concern of `filterClusters`. 

It would be better to not make the caller of `filterClusters` concerned with how to rearrange the data into a special format.

```suggestion
func filterClusters(clusters []mongodbatlas.Cluster, clusterNames []string, include bool) []mongodbatlas.Cluster {
	// Arrange cluster names for quick reference
	clusterNameSet := map[string]struct{}{}
	for _, clusterName := range clusterNames {
		clusterNameSet[clusterName] = struct{}{}
	}
	
	var filtered []mongodbatlas.Cluster
	for _, cluster := range clusters {
		if _, ok := clusterNameSet[cluster.Name]; (!ok && !include) || (ok && include) {
			filtered = append(filtered, cluster)
		}
	}
	return filtered
}
```